src/components/tree-view/tree-view.tsx

```diff
@@ -25,37 +28,77 @@ export interface TreeViewProps {
     selectedItem?: TreeItemData;
 }
 
+export interface TreeItemState {
+    isSelected: boolean;
+}
+
+export type StateMap = Map<TreeItemData, TreeItemState>;
+
 const itemIdPrefix = 'TREE_ITEM';
 
-export function TreeItem({ item, itemRenderer, onItemClick, isSelected }: TreeItemProps): JSX.Element {
+export function TreeItem({ item, itemRenderer, onItemClick, stateMap, state }: TreeItemProps): JSX.Element {
```

Comment:
in this case if we use `const TreeItem: React.StatelessComponent<TreeItemProps>` there would be no need to extend `TreeItemProps` with `React.Attributes` and no need to type the arguments